### ISSUE TITLE
after running multiple queries <*** KeyError: '@type'> starts to appear

### ISSUE THREAD

**Issue by kracwarlock** [Azure/azure-kusto-python#385]
#### Code Sample, a copy-pastable example if possible

```python
KCSB = KustoConnectionStringBuilder.with_aad_device_authentication(KUSTO_CLUSTER)
KCSB.authority_id = AAD_TENANT_ID
KUSTO_CLIENT = KustoClient(KCSB)
res = KUSTO_CLIENT.execute(KUSTO_DATABASE, query)
```
#### Problem description

This code is run in a loop with query being modified each time. After running successfully for a while I start getting this error
```python
*** KeyError: '@type'
```
I am suspecting that I am hitting some sort of rate limit. Can you confirm this and provide a better error message?

#### Output of ``pip freeze`` 
azure packages
```
azure-core==1.23.0
azure-identity==1.8.0
azure-kusto-data==3.1.0
azure-kusto-ingest==3.1.0
azure-storage-blob==12.10.0
azure-storage-queue==12.2.0
```

**Comment by woanware** [comment]
I have a similar scenario e.g. performing multiple queries, over multiple clusters, resulting in the same error as above

**Comment by AsafMah** [comment]
First of all - it's unclear from your phrasing, but if you're creating the client in a loop you should stop - you should be creating one client and re-using it for queries.

Secondly - I couldn't recreate the error - even when I managed to hit a rate limit, the error was properly parsed.

With that being said, I created #387 which will recover when failing to parse the error.

You could catch the exception and ignore it to resume your code,
And also I would appreciate if you send its contents to me, to see why it fails.

**Comment by kracwarlock** [comment]
So even after #387 I am still getting the same error. I found your alias and I will send you the script today.

**Comment by AsafMah** [comment]
Hey, we have released 3.1.1 with the PR in it,
Can you please run your code under it and say what you're getting?

**Comment by yihezkel** [comment]
Hi @kracwarlock,

@AsafMah believes this is fixed with the latest version, and confirmed in his environment. Since we haven't heard back from you in a month, we're going to close this issue. If you still experience this, feel free to reopen with repro details.
